### PR TITLE
perf: opt timeline structure

### PR DESCRIPTION
**Pull request by wumaolinmaoan**
## Summary by CodeRabbit

- **New Features**
  - Enhanced `CompositionComponent` with new methods for better control and manipulation.
  - Introduced `createVFXItem()` method in `VFXItem` class for streamlined VFX item creation.

- **Refactor**
  - Replaced `TimelineComponent` with `ObjectBindingTrack` across multiple components for improved consistency and functionality.
  - Reorganized import statements for better code readability and maintainability.

- **Bug Fixes**
  - Updated logic in various methods to ensure proper initialization and usage of properties.
  - Adjusted lifecycle and disposal logic in `calculate-item.ts` to improve performance and stability.

### CODE REVIEW

File: packages/effects-core/src/comp-vfx-item.ts [hunk 1]
```diff
@@ -231,4 +257,48 @@ export class CompositionComponent extends ItemBehaviour {
 
     return regions;
   }
+
+  // 深度优先遍历，创建排序后的数组
+  private dfsVFXItem (sortedArray: VFXItem<VFXItemContent>[], node: VFXItem<VFXItemContent>, childrenMap: Map<string, VFXItem<VFXItemContent>[]>): void {
```
Comment (yiiqii) [inline packages/effects-core/src/comp-vfx-item.ts:262]: dfs 是什么意思？

File: packages/effects-core/src/composition.ts [hunk 2]
```diff
@@ -397,7 +394,7 @@ export class Composition implements Disposable, LostHandler {
       this.restart();
     }
     // TODO: [1.31] @茂安 this.content.started 验证
```
Comment (yiiqii) [inline packages/effects-core/src/composition.ts:396]: 这个验证过了吗，没问题注释去掉吧

File: packages/effects-core/src/composition.ts [hunk 3]
```diff
@@ -633,10 +631,25 @@ export class Composition implements Disposable, LostHandler {
     }
   }
 
+  private toLocalTime (time: number) {
+    let localTime = time - this.rootItem.start;
+    const duration = this.rootItem.duration;
+
+    if (localTime - duration > 0.001) {
+      if (this.rootItem.endBehavior === spec.END_BEHAVIOR_RESTART) {
```
Comment (yiiqii) [inline packages/effects-core/src/composition.ts:639]: item 的 endBehavior 用 ItemEndBehavior 判断吧

File: packages/effects-core/src/utils/timeline-component.ts [hunk 4]
```diff
@@ -5,220 +5,4 @@ export interface TimelineComponentOptions {
   sizeOverLifetime?: spec.SizeOverLifetime,
```
Comment (yiiqii) [inline packages/effects-core/src/utils/timeline-component.ts:5]: TimelineComponentOptions 是不是没用了？

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: packages/effects-core/src/comp-vfx-item.ts (hunk 1 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/comp-vfx-item.ts:262]: 深度优先搜索

File: packages/effects-core/src/composition.ts (hunk 2 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/composition.ts]: 已去除

File: packages/effects-core/src/composition.ts (hunk 3 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/composition.ts]: 已修改

File: packages/effects-core/src/utils/timeline-component.ts (hunk 4 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/utils/timeline-component.ts]: 已删除

File: packages/effects-core/src/comp-vfx-item.ts [hunk 5]
```diff
@@ -231,4 +252,48 @@
...
+    // 第二步：从每个根节点开始深度优先遍历，并构建排序后的数组
+    const sortedArray: VFXItem<VFXItemContent>[] = [];
+
+    roots.forEach(root => {
+      // 对每个根节点及其子节点进行深度优先遍历
+      this.dfsVFXItem(sortedArray, root, childrenMap);
+    });
+
+    return sortedArray;
+  }
```
[the remark this answers is not in the document]
Comment (yiiqii) [reply packages/effects-core/src/comp-vfx-item.ts:298]: @wumaolinmaoan 看看机器人的建议

**Review by yiiqii — APPROVED** [review]

**Pull request merged**